### ISSUE TITLE
Different dimensions in output vs input

### ISSUE THREAD

**Issue by constantinpape** [bioimage-io/spec-bioimage-io#252]
Currently it's not clear how to specify outputs that have a different number of dimensions in the output than in the input when using implicit shapes.
E.g. a model that gets input (a) `bczyx` and outputs `bcyx`. Or (b) `bcyx` -> `bczyx`.
I would suggest that we set offset and scale to zero for (a) and scale to zero, offset to dim_size / 2 for (b).
This way we don't need to change the spec, but probably need to adapt the core lib so it can deal with these cases.

**Comment by FynnBe** [comment]
sounds good to me with one remark:
"offset to dim_size / 2 for (b)."; dim_size is not known for inputs defined with min,step. Any choice is somewhat arbitrary there, too, so maybe 0 would do the trick here, too.

**Comment by oeway** [comment]
Interesting question, it feels a bit unintuitive, and it won’t work it we
swapped the dimensions in the output. Can we add another key, say,
reference_dims, so we can explicitly set the reference dimension for
computing the outputs.

Even more flexible, considering we might want to compute differen dims with
respect to different input tensor, we can perhaps change the format of
reference tensor to a list with the format like: “input_tensor_name:y”. For
example if we have two inputs tensor and we want to define the outputs by
referring to both inputs, we can do something like this:

reference_dims: [ input-1:b, input-2:x, input-2:y, input-1:z]

If it’s a newly created dimension, then we set the reference dim to null.

**Comment by FynnBe** [comment]
For the straight forward case of only adding a dimension a `null` value could also be added to `scale`, with `offset` setting the length of the new dimension. 
e.g. 
```
reference_tensor: input-1
scale: [1, 1, 1, null]
offset: [0, 0, 0, 7]
```
this could be equivalent to specifying 
```
reference_dims: [input-1:b, input-1:x, input-1:y, null]
scale: [1, 1, 1, null]  # here we could also allow scale to be 0, but null is more explicit for a new dimension, so I would not allow 0 here to avoid confusion
offset: [0, 0, 0, 7]
```

Furthermore I would not allow to specify both, `reference_tensor` and `reference_dims`

**Comment by esgomezm** [comment]
Hi there, any news on this? @FynnBe do you think it could be implemented??

**Comment by FynnBe** [comment]
yes, but as discussed offline this does not have a high priority right now. (use of fixed output shape covers some of these use cases)

**Comment by FynnBe** [comment]
fixed soon by #521
